Docs/User Guide.md

```diff
@@ -0,0 +1,180 @@
+# VeriStand Communications Bus Template User Guide
+
+The template provided in this repo is intended to be used for creating VeriStand custom devices that implement communication protocols. This guide aims to provide some theory on the template design details on how to use the template to create a fully functional custom device.
+
+## Overview
+
+The main goal of this template is to enable custom device authors to implement the configuration and messaging for a given protocol only once. The protocol and hardware configurations are implemented separately to more easily enable supporting multiple vendors or hardware devices in a new custom device without having to start everything from a blank custom device template.
+
+Much of the overhead of creating custom devices is adapting LabVIEW code to correctly interface with VeriStand. The design of this template enables custom device authors to focus more on the code to implement a communication protocol and less on how VeriStand interacts with that code. The [Developer Guide](Developer%20Guide.md) provides more theory on the design of the template.
+
+## Creating a New Custom Device
+
+A utility for creating a new custom device from the template is located in this repo in the `Template Generator` directory. To generate a new custom device:
+
+1. Open the VI at `Template Generator/Template Tool/Clone Template Main.vi`.
+2. Follow the instructions on the VI's front panel to set the custom device name and output LabVIEW project location.
+3. Navigate to the **Destination Directory** provided in the VI and explore the custom device project.
+
+After the tool completes, the new custom device will be usable from VeriStand.
+
+## Theory of Operation
+
+The most flexible way of creating extensible architectures in LabVIEW is by using LabVIEW classes and interfaces. To create class or interface overrides, LabVIEW requires a new VI of the same name, with the same connector pane to exist. This presents challenges to creating extensible and reusable code pieces in custom devices because VeriStand requires top-level custom device libraries to be saved as `.llb` files. These files have a flat structure, which means two files of the same name can not exist in these file types.
```

Comment:
```suggestion
The most flexible way of creating extensible architectures in LabVIEW is by using LabVIEW classes and interfaces. To create class or interface overrides, LabVIEW requires a new VI of the same name, with the same connector pane to exist. This presents challenges to creating extensible and reusable code pieces in custom devices because VeriStand requires top-level custom device libraries to be saved as `.llb` files. These files have a flat structure, which means two files of the same name cannot exist in these file types.
```